packages/@lwc/aria-reflection/src/polyfill.ts

```diff
@@ -4,49 +4,18 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { hasOwnProperty, AriaPropNameToAttrNameMap } from '@lwc/shared';
-
-type NormalizedAttributeValue = string | null;
-type AriaPropMap = Record<string, NormalizedAttributeValue>;
-
-const nodeToAriaPropertyValuesMap: WeakMap<HTMLElement, AriaPropMap> = new WeakMap();
-
-function getAriaPropertyMap(elm: HTMLElement): AriaPropMap {
-    let map = nodeToAriaPropertyValuesMap.get(elm);
-
-    if (map === undefined) {
-        map = {};
-        nodeToAriaPropertyValuesMap.set(elm, map);
-    }
-
-    return map;
-}
-
-function getNormalizedAriaPropertyValue(value: any): NormalizedAttributeValue {
-    return value == null ? null : String(value);
-}
+import { AriaPropNameToAttrNameMap } from '@lwc/shared';
 
 function createAriaPropertyPropertyDescriptor(
     propName: string,
     attrName: string
 ): PropertyDescriptor {
     return {
         get(this: HTMLElement): any {
-            const map = getAriaPropertyMap(this);
-
-            if (hasOwnProperty.call(map, propName)) {
-                return map[propName];
-            }
```

Comment:
This `map` is a cute optimization (avoiding going through `setAttribute`/`getAttribute`/`removeAttribute`), but it breaks once we introduce the polyfill on `HTMLBridgeElement`, because the `this` (in `getAriaPropertyMap(this)`) is not necessarily consistent between the bridge (internal view of the element ) and `LightningElement` (external view).

I find it much simpler to just have the properties merely call `setAttribute`/`getAttribute`/`removeAttribute`.